charts/brigade/values.yaml

```diff
@@ -142,7 +160,7 @@ apiserver:
   service:
     ## If you're going to use an ingress controller, you can change the service
     ## type to CLusterIP.
-    type: NodePort
+    type: LoadBalancer
```

Comment:
> If you're going to use an ingress controller, you can change the service type to CLusterIP.

This remains true. imho, this is one of the best benefits of an ingress controller. You can get away with using a no-added-cost service type. (Compare to `LoadBalancer` which will cost you money in any of the public clouds.)

> I recall deploying with ingress + tls + cert manager (with tweaks similar to the rest of the changes in this PR) and not changing this value (apiserver.service.type), so it must've still been NodePort and it just worked.

This is also true. That would work, but isn't necessary.

> Also, do we want this default to change to LoadBalancer? Is this compatible with the default dev setup using minikube/kind?

minikube and kind won't complain, but they obviously won't give your service a public IP either. I _think_ a `LoadBalancer` service functionality is a superset of `NodePort` service functionality, but I really need to verify that. If I'm right, it's a sensible default.

If I'm wrong (equally likely!), then I'll amend this PR.

fwiw, I think you and I, more often than not install using `make hack` and we should note that _forces_ the service type to `NodePort` (e.g. `--set apiserver.service.type=NodePort`) because of the assumption you're developing locally and using either kind or or minikube. I mention this just because I managed to confused myself earlier today, stumbling multiple times over why I wasn't getting the service type I had asked for in my `values.yaml`.

Reply:
>  I think a LoadBalancer service functionality is a superset of NodePort service functionality, but I really need to verify that.

I have no verified this.

The only exception is that since service of type `LoadBalancer`, when on kind or minikube, will never be "ready" since it cannot obtain a public IP. This causes `helm install ... --wait` to hang indefinitely.

As I noted above, `make hack` and `make hack-deploy`, which we use extensively and which CI uses as well in the course of integration testing, _force_ the service type to `NodePort`. Due to this, there isn't really much of an issue.

Of course, it should be noted that our docs don't really exist yet and this quirk might stop us from using `helm install --wait` in the docs. If we decide later that this is a problem, we can always revisit what the default service type ought to be.

Does that sound ok?

Reply:
I still have a slight preference towards setting a default in the chart values that supports successful installs on any cluster (so, either NodePort or ClusterIP in this case), but fine with going this route for now and revisiting as needed in the future.

Reply:
> I still have a slight preference towards setting a default in the chart values that supports successful installs on any cluster

We both want the same thing, but it's leading us to different conclusions. So probably we need to define what is "success" in this case-- is it a smooth install only or is it a Brigade cluster that's ready to use? (ClusterIP or NodePort on a remote is unworkable without ingress or tricks accessible only to the operator-- like `kubectl port forward`.)

And I don't know the answer to this, so let's discuss offline.